### PR TITLE
optimize query

### PR DESCRIPTION
**Pull request by drhino**
no loop

### CODE REVIEW

File: products/workers/src/content/examples/extract-cookie-value.md [hunk 1]
```diff
@@ -18,25 +18,30 @@ tags:
...
+
+  // The cookie could not be found.
+  if (starts === -1)
+    return null
+
+  // Parse the cookie value.
+  const val = str.substr(starts + search.length, str.length),
+        end = val.indexOf(';')
+
+  return (end === -1) ? val : val.substr(0, end)
```
Comment (lukeed) [inline products/workers/src/content/examples/extract-cookie-value.md]:
```suggestion
function getCookie(request, key) {
  const cookie = request.headers.get('Cookie')
  
  // No cookie found
  if (!cookie) return

  // Search for the cookie key in the header.
  const search = `${key}=`
  const starts = cookie.indexOf(search)

  // The cookie could not be found.
  if (starts === -1) return

  // Parse the cookie value.
  const value = cookie.substring(starts + search.length, cookie.length)
  const end = value.indexOf(';')

  return end === -1 ? value : value.substring(0, end)
```

File: products/workers/src/content/examples/extract-cookie-value.md [hunk 2]
```diff
@@ -18,25 +18,30 @@ tags:
 const COOKIE_NAME = "__uid"
 
 /**
- * Gets the cookie with the name from the request headers
- * @param {Request} request incoming Request
- * @param {string} name of the cookie to get
+ * Returns a cookie value or null.
+ * @param {Request} request incoming request
+ * @param {string}  key of the cookie to get
+ * @returns {string} value of the cookie or null
```
Comment (lukeed) [inline products/workers/src/content/examples/extract-cookie-value.md]:
```suggestion
 * @returns {string|void} value of the cookie if found
```

**Review by lukeed — COMMENTED** [review]
Slight refactor to the code, main difference is checking for the header value once & returning `void` instead of `null` in the not-found conditions

**Review by kristianfreeman — APPROVED** [review]

**Comment by drhino** [comment]
@lukeed first sentence of comment block:
"Returns a cookie value or null." — still has the "null".

maybe: "Returns the value of a Cookie."

**Pull request merged**